src/shiver/views/sample.py

```diff
@@ -0,0 +1,585 @@
+
+        self.matrix_state_callback = None
+        self.lattice_state_callback = None
+        self.sample_data_callback = None
+        self.ub_matrix_from_lattice_callback = None
+        self.lattice_from_ub_matrix_callback = None
+
+    def start_dialog(self, name):
+        """initialize and start dialog"""
+        self.dialog = SampleDialog(name, parent=self)
+        self.dialog.setAttribute(Qt.WA_DeleteOnClose)
+        return self.dialog
+
+    def connect_ub_matrix_from_lattice(self, callback):
+        """callback for the matrix data"""
+        self.ub_matrix_from_lattice_callback = callback
+
+    def connect_lattice_from_ub_matrix(self, callback):
+        """callback for the matrix data"""
+        self.lattice_from_ub_matrix_callback = callback
+
+    def connect_sample_data(self, callback):
+        """callback for the matrix data"""
+        self.sample_data_callback = callback
+
+    def connect_matrix_state(self, callback):
+        """callback for the matrix data"""
+        self.matrix_state_callback = callback
+
+    def connect_lattice_state(self, callback):
+        """callback for the matrix data"""
+        self.lattice_state_callback = callback
+
+    # def connect_matrix_data(self, callback):
+    #    """callback for the matrix data"""
+    #    self.matrix_data_callback = callback
+
+    def connect_apply_submit(self, callback):
+        """callback for the apply submit button"""
+        self.btn_apply_callback = callback
+
+    def connect_load_submit(self, callback):
+        """callback for the loaf submit button"""
+        self.btn_load_callback = callback
+
+    def connect_nexus_submit(self, callback):
+        """callback for the nexus submit button"""
+        self.btn_nexus_callback = callback
+
+    def connect_isaw_submit(self, callback):
+        """callback for the isaw submit button"""
+        self.btn_isaw_callback = callback
+
+    def get_error_message(self, msg):
+        """received the error message from model"""
+        self.dialog.show_error_message(msg)
+
+
+class SampleDialog(QDialog):
+    """Histogram parameters widget"""
+
+    changed = Signal(dict)
+
+    def __init__(self, name, parent=None):
+        super().__init__(parent)
+
+        self.name = name
+        self.parent = parent  # define parent
+        layout = QVBoxLayout()
+        self.setLayout(layout)
+        self.setWindowTitle("UB setup")
+        self.setMinimumSize(QSize(630, 400))
+
+        # inputs
+        self.lattice_parameters = LatticeParametersWidget(parent, self)
+        layout.addWidget(self.lattice_parameters)
+
+        # loading buttons
+        self.load_btns = QWidget()
+        btn_layout = QHBoxLayout()
+        btn_layout.setSpacing(40)
+        # (first entry) from DAS logs
+        self.btn_load = QPushButton("Load UB")
+        self.btn_load.setFixedSize(QSize(100, 20))
+        btn_layout.addWidget(self.btn_load)
+
+        self.btn_nexus = QPushButton("Load Processed Nexus")
```

Comment:
"Load Processed Nexus" but it uses `LoadEventNexus` to load the workspace which would be unprocessed NeXus?

Reply:
All three file loading buttons have been updated with more appropriate names.